### ISSUE TITLE
Input for Model Architecture

### ISSUE THREAD

**Issue by tryggviy** [cszhangzhen/ANRL#6]
Hi,
After I read the paper, I suppose that you enable model to learn topology and attributes information at the same time by adding three loss functions together. However, I noticed that this model may not infer the topology structure well. Is it possible to improve it by adding one hot encoding value for model input(The model just use attributes as input)?

I am not quite clear about this part of your code        
def encoder(X):
            for i in range(self.layers - 1):
                name_W = 'encoder_W_' + str(i)
                name_b = 'encoder_b_' + str(i)
                X = tf.nn.tanh(tf.matmul(X, self.W[name_W]) + self.b[name_b])
            return X

Why bother using a for loop here, considering you only get the last value of this for loop as function return.

**Comment by cszhangzhen** [comment]
Yes, you can incorporate the structure information into the input.

The for loop here is used to build multilayer neural network for the encoder. The number of layers is controlled by the for loop, and using loop makes the code concise.

**Comment by tryggviy** [comment]
Thanks for your reply. Have you ever tried this one hot way to incorporate structure information into the input? If yes, did you see any difference in the performance with this method used.

**Comment by cszhangzhen** [comment]
No, I didn't.

It seems that there are lots of ways to incorporate the one-hot structure information. It depends on your choice.

**Comment by tryggviy** [comment]
Thanks for your reply.
